dashboards/google-kubernetes-engine/gke-optimization-signals.json

```diff
@@ -222,28 +223,126 @@
               "scale": "LINEAR"
             }
           }
-        },
-        "width": 3,
-        "xPos": 0,
-        "yPos": 2
+        }
       },
       {
-        "height": 2,
+        "xPos": 40,
+        "yPos": 0,
+        "width": 8,
+        "height": 8,
         "widget": {
+          "title": "Spot Node Usage",
           "scorecard": {
             "blankView": {},
             "thresholds": [],
             "timeSeriesQuery": {
               "timeSeriesQueryLanguage": "fetch k8s_node::kubernetes.io/node/cpu/core_usage_time\n| ${project_id}\n| ${location}\n| ${cluster_name}\n| group_by [],\n    sum(\n      if(metadata.user_labels.c'cloud.google.com/gke-spot' = 'true', val(), 0.0))\n    / .sum()\n| scale '%'",
               "unitOverride": ""
             }
-          },
-          "title": "Spot Discount Node"
+          }
+        }
+      },
+      {
+        "widget": {
+          "title": "Demand-Based Downscaling",
```

Comment:
Both new charts (Demand-Based Downscaling and Spot Node Usage) have issues:
- they don't filter by the template variables
- when filters are applied such that there is only one cluster being shown, the chart value doesn't match the scorecard value
- This chart is summing allocatable_utilization while the scorecard above is averaging allocatable_utilization - which one do you want?

Reply:
updated

Reply:
It looks like you've updated it to use `aggregate()` which is actually a sum, which seems wrong?

If I take away the aggregate() I see the individual nodes have allocatable utilization as follows:
![Screenshot 2024-03-18 at 10 25 55 AM](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/assets/6540967/577de4af-4a78-4391-929a-d7c1140c585c)

But it's wrong to sum this percentage over all of the nodes and say that "allocatable utilization" for the cluster is 927%...
![Screenshot 2024-03-18 at 10 27 53 AM](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/assets/6540967/059f3dca-f352-4ecb-9e78-e98586506c17)

While mean feels more correct:
![Screenshot 2024-03-18 at 10 28 47 AM](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/assets/6540967/c15ad8ae-53bc-4b61-b272-a1f339d9a501)